internal/pgd/module.go

```diff
@@ -74,10 +75,66 @@ const (
 	fmtPkg       = "fmt"
 	timePkg      = "time"
 	protozstdPkg = "github.com/pquerna/protoc-gen-dynamo/pkg/protozstd"
+	cryptoPkg    = "crypto/sha256"
 
 	timestampType = "google.protobuf.Timestamp"
 )
 
+// calculateShard computes the shard ID based on PK:SK using SHA256 hash
+func calculateShard(pkSk string, shardCount uint32) uint32 {
+	hash := sha256.Sum256([]byte(pkSk))
+	// Use first 4 bytes as uint32
+	hashValue := uint32(hash[0])<<24 | uint32(hash[1])<<16 | uint32(hash[2])<<8 | uint32(hash[3])
```

Comment:
Minor nit: this kinda thing is in the stdlib `binary.BigEndian.Uint32()`.

Reply:
Yeah, I wanted to not have to import that, but it's fine. I can add that.

Reply:
Basically this entire project is internal... I could expose this one function maybe? Feels weird to do that though.